### ISSUE TITLE
Jupyter kernel dies when ever i am using model.predict. did anyone face similar issue.

### ISSUE THREAD

**Issue by rt3722** [catboost/catboost#819]
Problem: {problem description}
catboost version: {version}
Operating System: {OS}
CPU: {CPU}
# GPU: {GPU}

**Comment by annaveronika** [comment]
Could you please provide a reproducing example? The code that you are running and data.

**Comment by rt3722** [comment]
please help.

This is the code i am running on Ubuntu 18.04 on google compute engine:

 
# coding: utf-8

# In[1]:


import pandas as pd
from sklearn.datasets import load_boston
import pyodbc
import numpy as np
from sklearn.model_selection import train_test_split
from catboost import CatBoostClassifier, FeaturesData, Pool, cv


# In[2]:


df = pd.read_csv('the_data_v2.txt', sep= ",",header=None)


# In[3]:


col_names = df.iloc[0]
df = df.drop([0], axis=0)
df.columns = col_names


# In[4]:


df = df.dropna(subset=['target'])


# In[5]:


Edition_c = df.Edition.value_counts()
Copyright_c = df.Copyright.value_counts()
Binding_c = df.Binding.value_counts()


# In[6]:


df['target'] = df['target'].astype('int64')


# In[7]:


df.target.value_counts()


# In[8]:


df['target'] = df['target'].astype('category')
df[['Publisher']] = df[['Publisher']].replace(np.NaN,"Missing_pub")


# In[ ]:


df['Running_drop_change_count'] = df['Running_drop_change_count'].astype('float32')
df['Running_no_change_count'] = df['Running_no_change_count'].astype('float32')
df['Running_positive_change_count'] = df['Running_positive_change_count'].astype('float32')
df['cumulative_increase'] = df['cumulative_increase'].astype('float32')
df['gap_months'] = df['gap_months'].astype('float32')
df['year_of_lpc_change'] = df['year_of_lpc_change'].astype('float32')


# In[ ]:



#'Running_drop_change_count', 'Running_no_change_count',
# 'Running_positive_change_count','lagged_1M'

X = df[['cumulative_increase', 'gap_months',
       'year_of_lpc_change', 'month_lpc', 'lagged_3M',
       'lagged_6M', 'lagged_12M', 'LPC_Start_Quarter', 'LPC_Start_Sem_Timeing',
       'Price_change_event_6M_Intervals', 'Publisher']]
y = df.iloc[:, [23]]


# In[ ]:


X_train1, X_test, y_train1, y_test = train_test_split(X, y, train_size=0.8, random_state=1234,stratify = y)
X_train, X_val, y_train, y_val = train_test_split(X_train1, y_train1, train_size=0.8, random_state=1234,stratify = y_train1)


# In[ ]:


model = CatBoostClassifier(loss_function = "MultiClass",
           eval_metric = "TotalF1",
           class_weights = [10,7,.1],
           od_pval = 0.0007,
           classes_count = 3,
           logging_level = "Verbose",
           snapshot_file='snapshot.bkp',
           snapshot_interval=600,
          max_ctr_complexity = 7
                          )
#'Running_drop_change_count', 'Running_no_change_count',

       
model.fit(X_train, y_train,
    cat_features=['month_lpc', 'lagged_3M', 'lagged_6M', 'lagged_12M','LPC_Start_Quarter',
       'LPC_Start_Sem_Timeing', 'Price_change_event_6M_Intervals',
       'Publisher'],
    eval_set=(X_val,y_val),
    verbose=True,
    plot=True
)


# In[ ]:


predictions = model.predict(X_test)

**Comment by annaveronika** [comment]
Please provide a minimal reproducing example - just the fit+predict that are failing. And we also need the data.

**Comment by rt3722** [comment]
fit is working.Only when i try to predict the kernel dies

**Comment by annaveronika** [comment]
We need a minimal 5 line example that we run and it fails. Could you do that?

**Comment by rt3722** [comment]
yup.sure than you

**Comment by rt3722** [comment]
for small data i am not getting any error

**Comment by annaveronika** [comment]
It's ok if you provide large data. But the code should be small, and it should use a ready dataset.

**Comment by rt3722** [comment]
My data has millions of records how to share it with you guys?

**Comment by rt3722** [comment]
One sec i will get it to you

**Comment by rt3722** [comment]
import pandas as pd
from sklearn.datasets import load_boston
import pyodbc
import numpy as np
from sklearn.model_selection import train_test_split
from catboost import CatBoostClassifier, FeaturesData, Pool, cv
import pickle

X_train = pd.read_pickle("./X_train.pkl")
X_val = pd.read_pickle("./X_val.pkl")
y_train = pd.read_pickle("./y_train.pkl")
y_val = pd.read_pickle("./y_val.pkl")
X_test = pd.read_pickle("./X_test.pkl")
y_test = pd.read_pickle("./y_test.pkl")


model = CatBoostClassifier(loss_function = "MultiClass",
           eval_metric = "TotalF1",
           class_weights = [10,7,.1],
           od_pval = 0.0007,
           classes_count = 3,
           logging_level = "Verbose",
           snapshot_file='snapshot.bkp',
           snapshot_interval=600,
          max_ctr_complexity = 7
                          )


       
model.fit(X_train, y_train,
    cat_features=['month_lpc', 'lagged_3M', 'lagged_6M', 'lagged_12M','LPC_Start_Quarter',
       'LPC_Start_Sem_Timeing', 'Price_change_event_6M_Intervals',
       'Publisher'],
    eval_set=(X_val,y_val),
    verbose=True,
    plot=True
)

predictions = model.predict(X_test)

**Comment by rt3722** [comment]
data link:

https://drive.google.com/file/d/1hNCxCi7R8b9ZDSOYpbdU2FpHyWkT-nFf/view?usp=sharing

**Comment by rt3722** [comment]
This is the only way i could figure out in "one sec"
Thank you. I would be really great full if you can help me out
I am running this code on google compute engine on Ubuntu 18.04

**Comment by rt3722** [comment]
NA

**Comment by Abhishekbhagwat** [comment]
Hi I am also facing this issue. But on a unix build
Dataset has around 75 features and approx 20 M records.

**Comment by snirbenyosef** [comment]
same here

**Comment by kizill** [comment]
Hello! It would be very helpful if you will share a code sample that will allow us to reproduce this crash.

**Comment by snirbenyosef** [comment]
@kizill 
Problem:
#### Describe the bug
I have a problem with predict func .
my dataframe Z_jor[something] is empty. And I try to use it in the predict func. I want it to continue but it just reset my kernel on jupiter notebook. I also  run my code from shell I get this error:
```python
No objects info loaded 
Floating point exception (core dumped)
```
#### Steps/Code to Reproduce

Example:
```python
import pandas as pd
import numpy as np
import pickle
import warnings
import matplotlib.pyplot as plt
import os
import datetime
from sklearn.preprocessing import LabelEncoder
from catboost import Pool, CatBoostRegressor

model_path = my model path
z_jor = some data frame 

regressor = pickle.load(open(model_path, 'rb'))['model']

regressor.predict(z_jor[something]) ## this crash the kernel
```

regressor:
```
print(regressor)
<catboost.core.CatBoostRegressor object at 0x7f689e3dcc18>
```



#### Expected Results
Not Crashing the kernel or show warrning  or clear error

#### Actual Results
1) from jupiter notebook>> just restart the kernel
2) from shell >>
```python
No objects info loaded 
Floating point exception (core dumped)
```

#### Versions
```
System:
    python: 3.6.9 (default, Jul 17 2020, 12:50:27)  [GCC 8.4.0]
executable: /usr/bin/python3
   machine: Linux-4.15.0-1083-azure-x86_64-with-Ubuntu-18.04-bionic

Python dependencies:
          pip: 20.1.1
   setuptools: 49.2.0
      sklearn: 0.23.1
        numpy: 1.19.1
        scipy: 1.5.2
       Cython: None
       pandas: 1.0.5
   matplotlib: 3.3.0
       joblib: 0.16.0
threadpoolctl: 2.1.0

```




catboost version: 0.23.2
Operating System: Linux-4.15.0-1083-azure-x86_64-with-Ubuntu-18.04-bionic
CPU: 8
# GPU:0

**Comment by AMilkov** [comment]
This happens when the shape[0] of the input data is zero, i.e. no rows are present just columns.
Still, the returned error should be more informative!
